src/cljx/core.clj

```diff
@@ -42,41 +45,62 @@ Returns a sequence of File objects, in breadth-first sort order."
   (if (empty? (str/trim code))
     code
     (-> (p/parser code)
-        z/xml-zip
+        (z/xml-zip)
         (walk rules)
-        z/root
-        sj/str-pt)))
+        (z/root)
+        (sj/str-pt))))
 
 (defn- relativize
-  [f root]
-  (-> root io/file .toURI (.relativize (-> f io/file .toURI))))
+  "Get relative path of filepath in respect
+  of rootpath."
+  [filepath rootpath]
+  (let [file-uri (.toURI (io/file filepath))
+        dest-uri (.toURI (io/file rootpath))]
+    (util/path (.relativize dest-uri file-uri))))
 
 (defn- destination-path
   [source source-path output-dir]
-  (.getAbsolutePath (io/file output-dir (str (relativize source source-path)))))
+  (let [^String destpath (relativize source source-path)  ; Calculate relative path.
+        ^URI destfile (io/file output-dir destpath)]      ; Create a dest file instance
+    (.getAbsolutePath destfile)))
+
+(defn- filter-files-for-path
+  "Filter a bunch of files that matches the
+  provided path."
+  [files sourcepath]
+  (let [^URI sourceuri (.toURI (io/file sourcepath))
+        ^String spath (util/path sourceuri)]
+    (filter (fn [f]
+              (let [^URI uri (.toURI f)
+                    ^String fpath (util/path uri)]
+                (.startsWith fpath sourcepath)))
```

Comment:
You are completely right. I thought that you says about spelling only... my mistake!